### ISSUE TITLE
Discussion: Add field to set tracking data retention time to more than 24h

### ISSUE THREAD

**Issue by kerel-fs** [glidernet/ogn-ddb#39]
The service
http://ognstats.ddns.net/igc/
currently provides IGC files of the last & current day.
Thanks @buczakwk for providing this service!!

As far as I can see, this respects our policy "You do not re-distribute OGN data older than 24 hours" (documented  [here](https://www.glidernet.org/ogn-data-usage/) since yesterday :see_no_evil: ) now. Discussion on this policy & license should happen on the ML, not here. :)

Imho it would be great for pilots to be able to download even older data when consent was given.
Do you have any proposals how to enable pilots to state their consent? Should this consent be given in the ogn-ddb (centralized), or in the service which provides the igc files?

**Comment by snip** [comment]
The feature is very cool yes. But ...

To allow to gives consent we need to be sure the pilot of this glider provided its consent.
And this seems far to be easy.
How to be sure the person who registered the device in ddb is the owner?
But he is also the pilote of this flight?
What to do in case of multiple owners?

Maybe we need to assigne the IGC flight to a person who provided its consent for this flight and not to the glider?
Maybe this can be done in link with Skylines which consider flight linked to pilot and not to glider?
See https://github.com/skylines-project/skylines/issues/419

Even if it is linked to Skylines and the account linked to the pilot agreed to have this information public. What proves it is really the pilote who provided this agreement?
With Skylines and XCSoar as tracker, the pilot has to configure XCSoar with the key provided by Skylines.
With OGN the ID is known so anyone can use it with fake Skylines account.
cc @Turbo87

Maybe after some users wants to have the IGC file available but maybe only for their own. Not publicly shared. Or shared with a group ...

About OGN DDB, my point of view is that we need to keep it as simple as possible. Today it is already complex for some people. But we can add some features on other sites and link it with DDB. Maybe by adding some oauth feature in DDB for example.

There is already some proposal (from Angel) about multiple tracker tracking same flight we need to deal with without adding too much complexity in DDB.
https://github.com/glidernet/ogn-ddb/issues/36

**Comment by wbuczak** [comment]
I'm very sorry its kind of difficult for me to participate in the
discussion, given I was twice kicked out from the OGN discussion group by
our self elected president;)

--------
sorry for typos, sent from my mobile phone

**Comment by ipearx** [comment]
Hi I'd like to re-vitalise this discussion. Not having this option is a real pain for PureTrack users. I have to explain they can't view or download their flight track history of OGN data greater than 1 day, while all other data sources they can. It's confusing why half or all of their tracks vanish.

It means I have to recommend people install another tracking system if they want their flight tracks to be available in the future.

The reality is if anyone wants to use the data longer term for 'bad' purposes, anyone can set up their own APRS client and store all the data as long as they want.

As for permission, yes we cannot confirm someone 'owns' their device, but they do tick a box to say they are the legal owner of the device.

So I'd like to propose the addition of an option to 'allow long term storage and redistribution of data' for each device.

**Comment by snip** [comment]
This data usage rule was conclusion of years of discussions.
https://www.glidernet.org/ogn-data-usage/
Unfortunately I don't think we can now be more permissive. We already reeived multiple time pressure from legal action with current situation.
This potential option in DDB is not enough as it not completly garanty the ownership of the device.

**Comment by ipearx** [comment]
Yeah I can appreciate that, but I'm not convinced it can't be changed. Have the legal threats actually amounted to anything?

I'd argue if the 'I own the device' tickbox isn't enough for 'store data for longer', then it's not really good enough for the other features of the database either.

The problem is it means I have to push pilots users to use other tracking systems that feed into PureTrack, e.g. dedicated cellular trackers, ADSB and not OGN...  And third party apps that share data to OGN I have to convince to send data to PureTrack directly rather than through OGN, to avoid the issue. I'd rather be promoting and supporting OGN, rather than getting people to work around it. I guess it is what it is!

**Comment by Turbo87** [comment]
arguably, if people wanted their own flights recorded then they should use IGC file loggers. those are not prone to Flarm reception issues either. am I missing something? 🤔

**Comment by pjalocha** [comment]
If  you need an IGC from the OGN then go to flightbook.glidernet.org within
24h hours of your flight: if you airfield is on the list, you can directly
download it from there.
If your airfield is not there, then arrange for it to be there: you need
your airfield description in .cup format.

Pawel.

**Comment by pjalocha** [comment]
Yet another method: ESP32 based trackers log data in internal flash, and
you can extract an IGC file.
THere are ready or almost ready devices now on the market, you simply buy
for 40-50 EUR like ThinkNode M5 which you simply upload with the code and
you have it ready and you transmit ADS-L as well, the most recent standard
by the EASA.

Pawel.

**Comment by ipearx** [comment]
There are many reasons why people/clubs/aircraft owners want to store tracking data for longer than 1 day.  And many reasons why using live tracking data is easier than using a logger (or the only option after the fact).

I do have an IGC download feature in PureTrack. I guess if I make an IGC upload feature, then people can download their OGN data, and re-upload it to be stored longer term? or they can upload to WeGlide or something. Then it's individuals breaching the OGN terms, not my website I guess?

So to recap at the moment:

- Anyone can set up an OGN APRS client to capture and store all OGN data for as long as they want.
- Anyone can download any OGN track data for up to 24 hours, and then re-upload it or store it forever.

So I don't think adding an option to say it's OK to store data longer breaches privacy any more than those existing options, it just makes it more convenient for those users who do want to store data in 3rd party apps for longer without faffing around with file downloads/uploads...

**Comment by snip** [comment]
> * Anyone can download any OGN track data for up to 24 hours, and then re-upload it or store it forever.

This one is not allowed. (edit: not allowed if it is redistributed)
But i do not see any issue when it is published on its own name: on weglide for its own flight as an example.
